### PR TITLE
Add hipSPARSE csrsort

### PR DESCRIPTION
**Pull request by upsj**
This PR adds hipSPARSE bindings for the *csrsort functions and the corresponding HIP kernels and tests.

### CODE REVIEW

File: hip/test/matrix/csr_kernels.hip.cpp [hunk 1]
```diff
@@ -112,6 +112,38 @@ class Csr : public ::testing::Test {
...
+                std::uniform_int_distribution<>(0, nnz_in_this_row - 1);
+            // shuffle `nnz_in_this_row / 2` times
+            for (size_t perm = 0; perm < nnz_in_this_row; perm += 2) {
+                const auto idx1 = swap_idx_dist(rand_engine);
+                const auto idx2 = swap_idx_dist(rand_engine);
+                std::swap(col_idx[idx1], col_idx[idx2]);
+                std::swap(vals[idx1], vals[idx2]);
+            }
+        }
+        auto local_mtx_cuda = Mtx::create(hip);
```
Comment (upsj) [inline hip/test/matrix/csr_kernels.hip.cpp]:
```suggestion
        auto local_mtx_hip = Mtx::create(hip);
```

File: hip/test/matrix/csr_kernels.hip.cpp [hunk 2]
```diff
@@ -112,6 +112,38 @@ class Csr : public ::testing::Test {
...
+        local_mtx_cuda->copy_from(local_mtx_ref.get());
```
Comment (upsj) [inline hip/test/matrix/csr_kernels.hip.cpp]:
```suggestion
        local_mtx_hip->copy_from(local_mtx_ref.get());
```

File: hip/test/matrix/csr_kernels.hip.cpp [hunk 3]
```diff
@@ -112,6 +112,38 @@ class Csr : public ::testing::Test {
...
+
+        return {std::move(local_mtx_ref), std::move(local_mtx_cuda)};
```
Comment (upsj) [inline hip/test/matrix/csr_kernels.hip.cpp]:
```suggestion
        return {std::move(local_mtx_ref), std::move(local_mtx_hip)};
```

File: hip/test/matrix/csr_kernels.hip.cpp [hunk 4]
```diff
@@ -112,6 +112,38 @@ class Csr : public ::testing::Test {
...
     }
 
+    struct matrix_pair {
+        std::unique_ptr<Mtx> ref;
+        std::unique_ptr<Mtx> hip;
+    };
+
+    matrix_pair gen_unsorted_mtx()
+    {
+        constexpr int min_nnz_per_row = 2;  // Must be larger/equal than 2
```
Comment (thoasm) [inline hip/test/matrix/csr_kernels.hip.cpp]: nit:
```suggestion
        constexpr int min_nnz_per_row = 2;  // Must be larger than/equal to 2
```

**Pull request merged**